tests/main/core18-configure-hook/task.yaml

```diff
@@ -30,3 +30,8 @@ execute: |
     fi
 
     test -e /var/snap/test-snapd-with-configure-core18/common/configure-ran
+
+restore: |
+    if snap list test-snapd-with-configure-core18; then
+        snap remove test-snapd-with-configure-core18
+    fi
```

Comment:
`snap remove` of an uninstalled snap succeeds, so the `if` isn't strictly necessary (which is why the other restores that don't do this check are OK).

Reply:
Ahh, that's neat. I didn't know that. I'll do a pass to simplify this.